Cargo.toml

```diff
@@ -129,9 +129,9 @@ winreg = "0.7"
 # wasm
 
 [target.'cfg(target_arch = "wasm32")'.dependencies]
-js-sys = "0.3.28"
-wasm-bindgen = { version = "0.2.51", features = ["serde-serialize"] }
-wasm-bindgen-futures = "0.4.1"
+js-sys = "0.3.45"
+wasm-bindgen = { version = "0.2.68", features = ["serde-serialize"] }
+wasm-bindgen-futures = "0.4.18"
```

Comment:
Are these version bumps required? Is it only with these newer versions that we can check for `ServiceWorkerGlobalScope`?

Reply:
Technically, it does compile without those version bumps, but here's some additional info:
When you run "wrangler generate --type rust" (wrangler is the utility cloudflare provides that generates an empty project template), it generates a Cargo.toml with
  wasm-bindgen = "=0.2.65"
I didn't find any documentation on CF  used 0.2.65 specifically. On different projects I've used 0.2.65, and 0.2.68 without problem. The version in reqwest's previous Cargo.toml, 0.2.51, is from Sept 2019 which seems like eons ago in wasm time.
Since these crates (js-sys, wasm-bindgen, wasm-bindgen-futures) are so tightly linked, I picked versions that  were all committed at about the same time ~Sept 9. Not the absolute latest, but based on minor changes since, it appears stable. The set of versions checked in is the set I've been testing with.

(Related - are you open to a pull request that adds ClientBuilder.default_headers support in wasm?)

Thanks!

Reply:
For the record, I attempted to use JsCast's instanceof and dyn_into, using both Window and ServiceWorkerGlobalScope, but in one platform or the other always each of these hit either a panic or a javascript exception, which is hard to catch in Rust. The Reflect method seems to be the cleanest and safest approach. I did not try to measure performance, but given the number of other rust/js method calls prior to fetch, and the fact that fetch is a network operation, I am making an assumption that the performance cost of the Reflect test is negligible for both code paths.